### PR TITLE
fix: resolve linting issues, enable CI

### LINKED ISSUE #1: fix lint warnings and errors so that npm run test works as expected

### PR DESCRIPTION
**Pull request by fspoettel**
closes #1 

most of the changes were done by the (very opinionated) linter. see the following list of linting issues that were fixed manually:

Screenshot 2023-07-13 at 14 09 45

### CODE REVIEW

File: source/js/timezone-fixers/tiktok.js [hunk 1]
```diff
@@ -12,45 +12,59 @@ const attachedElements = new Set();
  * 	  HARD TODO: is it possible to do it for the "you may like" section on the right side of a video page? there is no id in the html but perhaps the click event listener could somehow be intercepted/mocked.
  */
 async function attachTimeInfo() {
-	// for a maximized video page -> hover over username
-	Array.from(document.querySelectorAll('span[data-e2e="browser-nickname"]')).map(span => {
-		if (attachedElements.has(span)) return; // skip already processed
+	// For a maximized video page -> hover over username
+	for (const span of Array.from(document.querySelectorAll('span[data-e2e="browser-nickname"]'))) {
```
Comment (msramalho) [inline source/js/timezone-fixers/tiktok.js:16]: curiosity: are there strict rules against using .map in this context? I usually find it easier to read/code.

Comment (fspoettel) [reply source/js/timezone-fixers/tiktok.js:16]: So the change I did was `.map()` to `.forEach()`. `.map()` expects a return value and no side-effects, which engines optimize around.`.forEach()` is clearer for cases where you don't want to return anything, as is the case here.

The change to `for ... of ...` was then done by the linter.

Comment (fspoettel) [reply source/js/timezone-fixers/tiktok.js:16]: see https://github.com/microsoft/TypeScript/blob/main/src/lib/es5.d.ts#L1235-L1246

**Review by msramalho — APPROVED** [review]
reviewed and tested :heavy_check_mark:

**Pull request merged**